src/lib.rs

```diff
@@ -1936,22 +1941,30 @@ pub unsafe trait FromBytes: FromZeros {
     ///     checksum: [u8; 2],
     /// }
     ///
-    /// // These are more bytes than are needed to encode a `PacketHeader`.
-    /// let bytes = &[0, 1, 2, 3, 4, 5, 6, 7, 8, 9][..];
+    /// #[derive(FromBytes, KnownLayout, NoCell)]
+    /// #[repr(C)]
+    /// struct Packet {
+    ///     header: PacketHeader,
+    ///     body: [[u8; 2]],
+    /// }
     ///
-    /// let header = PacketHeader::ref_from_prefix(bytes).unwrap();
+    /// // These are more bytes than are needed to encode a `Packet`.
+    /// let bytes = &[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14][..];
     ///
-    /// assert_eq!(header.src_port, [0, 1]);
-    /// assert_eq!(header.dst_port, [2, 3]);
-    /// assert_eq!(header.length, [4, 5]);
-    /// assert_eq!(header.checksum, [6, 7]);
+    /// let packet = Packet::ref_from_prefix(bytes).unwrap();
+    ///
+    /// assert_eq!(packet.header.src_port, [0, 1]);
+    /// assert_eq!(packet.header.dst_port, [2, 3]);
+    /// assert_eq!(packet.header.length, [4, 5]);
+    /// assert_eq!(packet.header.checksum, [6, 7]);
+    /// assert_eq!(packet.body, [[8, 9], [10, 11], [12, 13]]);
     /// ```
     #[inline]
     fn ref_from_prefix(bytes: &[u8]) -> Option<&Self>
```

Comment:
Need to figure out how to reword this paragraph from the doc comment:

```rust
    /// `ref_from_prefix` returns a reference to the first `size_of::<Self>()`
    /// bytes of `bytes`. If `bytes.len() < size_of::<Self>()` or `bytes` is not
    /// aligned to `align_of::<Self>()`, this returns `None`.
```

Reply:
Added a TODO in #29 to block release on updating this doc comment, so we don't have to resolve this now.